Add 1.0 Docker images, and upgrade all the images to follow the Quick Starts for each language.

markmandel opened the pull request:
I updated the language version for each one as well to what is latest.

I was tempted to remove the onbuild images - simply because they expanded the support required, but I updated the version numbers they depended on and left them in. Let me know your thoughts.

Please let me know if there is any adjustment / refinement you would like to see.

markmandel committed:
Update Docker Images to gRPC 1.0

This brings all the Docker images up to date with the 1.0 release of
gRPC.

It also aligns each of the images with the steps taken in the
Quick Starts, so they can be used as temporary environments to follow
along with the Quick Starts or as an additional reference.

nathanielmanistaatgoogle commented on 1.0/python/README.md:
```diff
@@ -0,0 +1,48 @@
+# gRPC Python Docker image
+
+This is the official docker image for the Python facility of grpc.  For an
```
gRPC when written in prose should always be styled as "gRPC".

nathanielmanistaatgoogle commented on 1.0/python/README.md:
```diff
@@ -0,0 +1,48 @@
+# gRPC Python Docker image
+
+This is the official docker image for the Python facility of grpc.  For an
+overview and usage examples, see the grpc python documentation.
```
Styling here should be "gRPC Python".

nathanielmanistaatgoogle commented on 1.0/python/Dockerfile:
```diff
@@ -0,0 +1,34 @@
+
+ENV GRPC_PYTHON_VERSION 1.0.0
+RUN python -m pip install --upgrade pip
+RUN pip install grpcio==${GRPC_PYTHON_VERSION} grpcio-tools==${GRPC_PYTHON_VERSION}
```
Missing newline at end-of-file (here and in at least one other file in this pull request).

markmandel commented:
Updated the Dockerfiles so that they all have linebreaks, and fixed up all the readme's in the `1.0` folder such that they state "gRPC" (worth nothing that in `0.10` and `0.11` folders, they are still "grpc")

Please let me know if you have any further feedback.

nathanielmanistaatgoogle commented on 1.0/python/README.md:
```diff
@@ -0,0 +1,48 @@
+# gRPC Python Docker image
+
+This is the official docker image for the Python facility of gRPC.  For an
+overview and usage examples, see the gRPC python documentation.
```
"gRPC Python" on this line too.

nathanielmanistaatgoogle commented:
I was just jumping on some cosmetic stuff; @nicolasnoble will be the reviewer of substance for this change (or he will route this change to someone else who will be).

markmandel commented:
No worries. Good catches. Fixed up now. :+1:

nicolasnoble commented on 1.0/cxx/Dockerfile:
```diff
@@ -0,0 +1,50 @@
+
+#install protoc
+RUN cd /var/local/git/grpc/third_party/protobuf && \
+    make && make install && make clean
```
Missing end of line.

nicolasnoble reviewed (changes requested):
One tiny nit, otherwise lgtm.

markmandel commented:
Added newline. Let me know if you would like a rebase to clean up the commit history or not :smiley:

nathanielmanistaatgoogle commented:
I favor this squashing; @nicolasnoble will be the one to actually decide.

markmandel commented:
@nicolasnoble What do you reckon? Shall I squash it, and then it can be merged? :smile:

nicolasnoble commented:
I don't really care either way - that repository has less importance when it comes to history, so squash, don't squash, do what you prefer :)

markmandel commented:
@nathanielmanistaatgoogle Do you feel strongly either way? I don't mind. I just want to get this through the door? :smile:

nathanielmanistaatgoogle commented:
Squash.

nathanielmanistaatgoogle commented:
Also for extra bonus points adhere to [these guidelines](http://chris.beams.io/posts/git-commit/#seven-rules).

markmandel commented:
@nathanielmanistaatgoogle Squashed and updated the commit message.

(Love that guideline. Stealing for future use)

markmandel commented:
Thank you!
